tests/migrations/test_autodetector.py

```diff
@@ -2663,16 +2663,83 @@ class Migration(migrations.Migration):
+        self.assertEqual(migration.suggest_name(), 'initial')
+
+    def test_empty(self):
+        class Migration(migrations.Migration):
+            operations = []
 
     def test_none_name(self):
         class Migration(migrations.Migration):
             operations = [migrations.RunSQL('SELECT 1 FROM person;')]
+        migration = Migration("some_migration", "test_app")
+        self.assertTrue(migration.suggest_name().startswith("auto_"))
 
         migration = Migration('0001_initial', 'test_app')
         suggest_name = migration.suggest_name()
         self.assertIs(suggest_name.startswith('auto_'), True)
 
+    def test_no_fragments(self):
+        class Migration(migrations.Migration):
+            operations = [migrations.RunSQL("SELECT 1")]
+        migration = Migration('some_migration', "test_app")
+        self.assertTrue(migration.suggest_name().startswith("auto_"))
+
+    def test_single_create_model(self):
+        class Migration(migrations.Migration):
+            operations = [migrations.CreateModel("Person", fields=[])]
+        migration = Migration("some_migration", "test_app")
+        self.assertEqual(migration.suggest_name(), "person")
+
+    def test_two_operations(self):
+        class Migration(migrations.Migration):
+            operations = [
+                migrations.CreateModel("Person", fields=[]),
+                migrations.DeleteModel("Animal"),
+            ]
+
+        migration = Migration("some_migration", "test_app")
+        self.assertEqual(migration.suggest_name(), "person_delete_animal")
+
+    def test_many_create_models(self):
+        class Migration(migrations.Migration):
+            operations = [
+                migrations.CreateModel("Person1", fields=[]),
+                migrations.CreateModel("Person2", fields=[]),
+                migrations.CreateModel("Person3", fields=[]),
+                migrations.CreateModel("Person4", fields=[]),
+                migrations.CreateModel("Person5", fields=[]),
+                migrations.CreateModel("Person6", fields=[]),
+                migrations.CreateModel("Person7", fields=[]),
+                migrations.CreateModel("Person8", fields=[]),
+            ]
+        migration = Migration("some_migration", "test_app")
+        self.assertEqual(
+            migration.suggest_name(),
+            "person1_person2_person3_person4_person5_person6_and_more",
+        )
+
+    def test_operation_with_no_suggested_name(self):
+        class Migration(migrations.Migration):
+            operations = [
+                migrations.CreateModel("Person1", fields=[]),
+                migrations.RunSQL('SELECT 1 FROM person;'),
+            ]
+        migration = Migration("some_migration", "test_app")
+        self.assertIs(migration.suggest_name().startswith('auto_'), True)
+
+    def test_single_create_model_long_name(self):
+        """
+        Check that the first migration_name_fragment values isn't cut when
```

Comment:
To add more explanation, docstrings should state the expected behavior and omit prefixes like "Check that" since all tests check things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).